src/Controller/Admin/OrderController.php

```diff
@@ -0,0 +1,157 @@
+use Doctrine\ORM\EntityManagerInterface;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+use App\Repository\OrderRepository;
+use App\Repository\ProductRepository;
+use App\Repository\OrderProductRepository;
+use App\Entity\Order;
+use App\Entity\OrderProduct;
+use App\Entity\Product;
+use App\Form\SellerOrderType;
+use App\Form\AdminOrderType;
+use App\Form\AddProductToOrderType;
+use App\Manager\CartManager;
+use Symfony\Component\Form\FormError;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+/**
+ * @Route("/order")
+ */
+class OrderController extends AbstractController
+{
+    private $requestStack;
+
+    public function __construct(RequestStack $requestStack)
+    {
+        $this->requestStack = $requestStack;
+    }
+
+    private function getSession(): SessionInterface
+    {
+        return $this->requestStack->getSession();
+    }
+
+    /**
+     * @Route("/", name="admin_order_index", methods={"GET"}, host="admin.%domain%")
+     * @IsGranted("ROLE_ADMIN")
+     */
+    public function index(OrderRepository $orderRepository, OrderProductRepository $orderProductRepository): Response
+    {
+        return $this->render('admin/order/index.html.twig', [
+            'orders' => $orderRepository->findByNot('status', Order::STATUS_CART),
+        ]);
+    }
+
+    /**
+     * @Route("/{id}", name="admin_order_show", methods={"GET"}, host="admin.%domain%")
+     * @IsGranted("ROLE_ADMIN")
+     */
+    public function show(Order $order): Response
+    {
+        return $this->render('admin/order/show.html.twig', [
+            'order' => $order,
+        ]);
+    }
+
+    /**
+     * @Route("/{id}/edit", name="admin_order_edit", methods={"GET","POST"}, host="admin.%domain%")
+     * @IsGranted("ROLE_ADMIN")
+     */
+    public function edit(Request $request, Order $order, EntityManagerInterface $entityManager, OrderProductRepository $orderProductRepository, ProductRepository $productRepository): Response
+    {
+        if (null == $this->getSession()->get('clonedOrder')) {
+            $this->getSession()->set('clonedOrder', $order);
+        }
+        
+        $form = $this->createForm(AdminOrderType::class, $order);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $clonedOrder = $this->getSession()->get('clonedOrder');
+            $this->getSession()->remove('clonedOrder');
+            $updatedOrderProducts = $form->get('orderProducts')->getData()->toArray();
+
+            if (isset($status)) {
+                $order->setStatus($status);
+                
+                $orderProducts = $order->getOrderProducts();
+    
+                foreach ($orderProducts as $orderProduct) {
+                    $orderProduct->setStatus($status);
+                }
+            }
+            
+            foreach ($updatedOrderProducts as $updatedOrderProduct) {
```

Comment:
Please move this code in service. currently the controller is getting bloated.

Reply:
done